src/main/java/org/ehrbase/fhirbridge/ehr/converter/BodyHeightCompositionConverter.java

```diff
@@ -31,29 +34,34 @@ public KorpergrosseComposition toComposition(Observation observation) {
             return null;
         }
 
-        // references:
-        // https://ckm.highmed.org/ckm/templates/1246.169.1038
-        // https://tools.openehr.org/designer/?code=39ddef9b1e993dc441ff#/designer/repos/num-2021-01-11
-        // https://simplifier.net/ForschungsnetzCovid-19/BodyHeight/~overview
-        // https://simplifier.net/forschungsnetzcovid-19/observation-example-body-height
-
         KorpergrosseComposition result = new KorpergrosseComposition();
         GrosseLangeObservation grosseLangeObservation = new GrosseLangeObservation();
 
-        DateTimeType fhirEffectiveDateTime;
+
+        ZonedDateTime fhirEffectiveDateTime = null;
         try {
-            // default for every observation
-            fhirEffectiveDateTime = observation.getEffectiveDateTimeType();
 
-            //BSa Wie erscheint das im Mapping? Und wo finde ich das Referenzmodell mit diesen Infos zu openEHR?
-            grosseLangeObservation.setTimeValue(fhirEffectiveDateTime.getValueAsCalendar().toZonedDateTime());
-            grosseLangeObservation.setOriginValue(fhirEffectiveDateTime.getValueAsCalendar().toZonedDateTime()); // mandatory
+            if(observation.hasEffectiveDateTimeType()) {
+
+                // default for every observation
+                fhirEffectiveDateTime = observation.getEffectiveDateTimeType().getValueAsCalendar().toZonedDateTime();
+                grosseLangeObservation.setTimeValue(fhirEffectiveDateTime);
+                grosseLangeObservation.setOriginValue(fhirEffectiveDateTime);
+
+            } else if(observation.hasEffectivePeriod()) {
+                fhirEffectiveDateTime = observation.getEffectivePeriod().getStart().toInstant().atZone(ZoneId.systemDefault());
+                grosseLangeObservation.setTimeValue(fhirEffectiveDateTime);
+                grosseLangeObservation.setOriginValue(fhirEffectiveDateTime);
+            } else {
+                throw new UnprocessableEntityException("No time is set");
+            }
+
             grosseLangeObservation.setLanguage(Language.DE); // FIXME: we need to grab the language from the template
             grosseLangeObservation.setSubject(new PartySelf());
 
             // special mapping content
             grosseLangeObservation.setGrosseLangeUnits(observation.getValueQuantity().getCode());
-            //BSa Hier ist nicht sichtbar, ob Körpergröße oder Geburt gesetzt wird -> woher die Info?
+
             grosseLangeObservation.setGrosseLangeMagnitude(observation.getValueQuantity().getValue().doubleValue());
 
         } catch (Exception e) {
```

Comment:
Do you really want to catch every exception here or just specific ones ?

Reply:
That was overtaken from our "gold standard" example body weight. I only would forward the exception, so I think I can remove the try-catch block. What do you think?